src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/bundle/loader/ListenPortLoader.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+package com.ca.apim.gateway.cagatewayconfig.tasks.zip.bundle.loader;
+
+import com.ca.apim.gateway.cagatewayconfig.tasks.zip.beans.Bundle;
+import com.ca.apim.gateway.cagatewayconfig.tasks.zip.beans.ListenPort;
+import com.ca.apim.gateway.cagatewayconfig.tasks.zip.beans.ListenPort.ListenPortTlsSettings;
+import com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentTools;
+import org.w3c.dom.Element;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.ca.apim.gateway.cagatewayconfig.tasks.zip.beans.ListenPort.ClientAuthentication.fromType;
+import static com.ca.apim.gateway.cagatewayconfig.util.gateway.BuilderUtils.mapPropertiesElements;
+import static com.ca.apim.gateway.cagatewayconfig.util.gateway.BundleElementNames.*;
+import static com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentTools.*;
+import static java.lang.Integer.parseInt;
+
+public class ListenPortLoader implements BundleEntityLoader {
+
+    private final DocumentTools documentTools;
+
+    ListenPortLoader(DocumentTools documentTools) {
+        this.documentTools = documentTools;
+    }
+
+    @Override
+    public void load(final Bundle bundle, final Element element) {
+        final Element listenPortElement = documentTools.getSingleChildElement(documentTools.getSingleChildElement(element, RESOURCE), LISTEN_PORT);
+
+        final String name = documentTools.getSingleChildElement(listenPortElement, NAME).getTextContent();
+        final String protocol = documentTools.getSingleChildElement(listenPortElement, PROTOCOL).getTextContent();
+        final Integer port = parseInt(documentTools.getSingleChildElement(listenPortElement, PORT).getTextContent());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Change this instance-reference to a static reference. ([squid:S2209](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS2209))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=62&issues=AWVuUH36XKENQn1Sripb&open=AWVuUH36XKENQn1Sripb)

[ISSUE_KEY]: AWVuUH36XKENQn1Sripb